From 8e83802793575c900a91bd663eb6536ffbe97a5e Mon Sep 17 00:00:00 2001
From: Nikita <nikich340@gmail.com>
Date: Tue, 23 Feb 2016 15:19:23 +1000
Subject: [PATCH 083/211] Snap: Fix incorrect preview layout surface size in
 landscape mode

If height is smaller than width it means that orientation sets to horizontal
and we should decrease width instead height to get normal preview size.
Change-Id: Ifa211ad2604fb730fb85b01ae1d7c93c9b7aa2ce
---
 src/com/android/camera/PhotoUI.java | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/src/com/android/camera/PhotoUI.java b/src/com/android/camera/PhotoUI.java
index 6a9e8aa51..3bd318195 100755
--- a/src/com/android/camera/PhotoUI.java
+++ b/src/com/android/camera/PhotoUI.java
@@ -395,8 +395,13 @@ public class PhotoUI implements PieListener,
         } else {
             float width = mMaxPreviewWidth, height = mMaxPreviewHeight;
             if (width == 0 || height == 0) return;
-            if(mScreenRatio == CameraUtil.RATIO_4_3)
-                height -=  (mTopMargin + mBottomMargin);
+            if (mScreenRatio == CameraUtil.RATIO_4_3) {
+                if (height > width) {
+                    height -= (mTopMargin + mBottomMargin);
+                } else {
+                    width -= (mTopMargin + mBottomMargin);
+                }
+            }
             if (mOrientationResize) {
                 scaledTextureWidth = height * mAspectRatio;
                 if (scaledTextureWidth > width) {
@@ -415,7 +420,7 @@ public class PhotoUI implements PieListener,
                         scaledTextureHeight = height;
                     }
                 } else {
-                    if(Math.max(height, width * mAspectRatio) > height) {
+                    if (Math.max(height, width * mAspectRatio) > height) {
                         scaledTextureWidth = height / mAspectRatio;
                         scaledTextureHeight = height;
                     } else {
@@ -436,7 +441,7 @@ public class PhotoUI implements PieListener,
                 lp = new FrameLayout.LayoutParams((int) scaledTextureWidth,
                         (int) scaledTextureHeight, Gravity.CENTER);
             }
-            if(mScreenRatio == CameraUtil.RATIO_4_3) {
+            if (mScreenRatio == CameraUtil.RATIO_4_3) {
                 lp.gravity = Gravity.CENTER_HORIZONTAL | Gravity.TOP;
                 lp.setMargins(0, mTopMargin, 0, mBottomMargin);
             }
-- 
2.17.1

